x/programs/runtime/map.go

```diff
@@ -42,14 +41,18 @@ func NewMapModule(log logging.Logger, meter Meter) *MapModule {
 	return &MapModule{
 		meter: meter,
 		log:   log,
-		store: storage{
-			state:   make(map[uint64]maps),
-			mods:    make(map[uint64]api.Module),
-			counter: 0,
-		},
 	}
 }
 
+// GlobalStorage is a global variable that holds the state of all programs.
+// This is a placeholder storage system intended to show how a wasm program would access/modify persistent storage.
+// Needs to be gloabl, so state can be persisted across multiple runtime intances.
+var GlobalStorage = storage{
+	state:    make(map[int64]maps),
+	counter:  0,
+	Programs: make(map[uint32][]byte),
```

Comment:
as noted above programs should be stored in a db on create and retrieved via db get in `program_invoke`.